### ISSUE TITLE
Ctrlr conflicts with External Instrument in Ableton

### ISSUE THREAD

**Issue by Stakker** [RomanKubiak/ctrlr#143]
Hello,

sorry if this is the wrong place for this, but:

When I place Ctrlr (AU) on an Ableton track with an External Instrument device (the default way Ableton handles external instruments), it overwrites the device. And vice versa. I cannot place Ctrlr and External Instrument on the same track, which kinda defeats the purpose of the whole enterprise.

I guess I could create a new track with the same MIDI channel and put Ctrlr there, but that would make navigating the tracks really clunky. I have about 8 hardware synth tracks so I really don't want 8 extra tracks to mess up my projects.

I hope this is just some sort of miscategorization of the plugin, and not a deeper issue? I can (obviously) place all my other AU plugins on tracks with External Instrument.

The Ctrlr panel seems to work fine BTW, so that's good. Knobs are updated fine. There's just no audio :-P I'm using the latest Ableton 10.

Cheers!

**Comment by rkubiak01** [comment]
does the same happen when you try to use the VST version of ctrlr ?

**Comment by Stakker** [comment]
Same thing with the VST. Ableton seems to classify it as an instrument, and a track can only have one instrument. (I have no idea what Ableton actually technically does though.)

BTW, you have to resort to running xattr in terminal to get the plugin to work with Gatekeeper... I guess that's just how it is these days without an Apple developer account & notarization, so I can personally live with that. Less nerdy users will probably just give up when they see the error, though :-/

**Comment by Godlike-Productions** [comment]
What would be amazing is if ctrlr could return audio in the same way as an external instrument.  I know it pushes it outside of the MIDI envelope, but it would simplify using panels on pretty much all platforms.

**Comment by MatjoeP** [comment]
You peobably can walkaround by using another channel. Not ideal though. 
Not here to brag or convert you, but that is possible in bitwig studio. 
We got two nested containers built into the native instrument devices. One for midi fx one for audio fx. So you can save  presets containing midi fx and audio fx including third party vsts etc.  Pretty neat.

**Comment by Stakker** [comment]
Yeah Bitwig seems cool. Ableton can also have a free mix of midi and audio fx/devices on each track, but only one instrument (the device that receives the midi input configured for the track). And for some reason it thinks Ctrlr is an instrument. Or at least that’s how it seems to me :-)
